### ISSUE TITLE
hashing is not consistent for thunks

### ISSUE THREAD

**Issue by ceedubs** [unisonweb/unison#2923]
In the following example, `'(7)` is hashed twice and returns different results each time.

```
> let
  hash = crypto.hash Blake2b_256
  (hash '(7), hash '(7))

          ⧩
          (0xsfaab5fa37fa0b5468eb6a67c2be02f1746c9783d2b1375ccc6143801094303ff,
           0xsfddff319d0d5dcb4728521b19582cb56739626b2b634738948426fc2142c0ca9)
```

The hashes seem to be stable across runs which suggests to me that the hash includes information about where the argument is in the surrounding closure context or something like that.

**Comment by ceedubs** [comment]
This is closely related to #2163 and we should check whether it's resolved by #2615.

**Comment by ceedubs** [comment]
Resolved by #2615
